### PR TITLE
fix pinning for injected provider

### PR DESCRIPTION
**Pull request by Aniket-Engg**

### CODE REVIEW

File: libs/remix-ui/run-tab/src/lib/actions/events.ts [hunk 1]
```diff
@@ -14,6 +14,10 @@ import { shortenAddress } from "@remix-ui/helper"
 const _paq = window._paq = window._paq || []
 
 export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
+  const currentNetwork = {
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/events.ts:17]: could you add a comment in a code on why you need to do this?

File: libs/remix-ui/run-tab/src/lib/actions/events.ts [hunk 2]
```diff
@@ -33,7 +37,6 @@ export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
     }
     setFinalContext(plugin, dispatch)
     fillAccountsList(plugin, dispatch)
-    await loadPinnedContracts(plugin, dispatch)
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/events.ts:36]: what happens if one change the network in metamask (or any injected provider) ?

Comment (yann300) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:36]: > what happens if one change the network in metamask (or any injected provider) ?

I mean that the network get changed not by selecting a new env in udapp, but by selecting another network directly from metamask.

Comment (Aniket-Engg) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:36]: both the cases are handled by this condition `if (currentNetwork.provider !== networkProvider() || (!isVM && currentNetwork.chainId !== network.id))` you can try it on netlify link

File: libs/remix-ui/run-tab/src/lib/actions/events.ts (hunk 1 above)
Comment (Aniket-Engg) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:20]: done

**Review by yann300 — APPROVED** [review]

**Pull request merged**